### PR TITLE
feat(proxy): support environment http_proxy and https_proxy

### PR DESCRIPTION
**Pull request by errorx666**

### CODE REVIEW

**Comment by errorx666** [comment]
I have signed the CLA.

**Comment by dignifiedquire** [comment]
Thank you. Could you give some background on why this should be added to karma directly?

**Comment by errorx666** [comment]
@dignifiedquire
I was trying to proxy a remote URL from Karma, but it was unable to resolve the remote hostname because it was not reading my system's proxy settings correctly.  This fixes that issue.
I'm not sure that it would be possible to add it externally as the proxy object is not exposed or exported by the API.  Also, reading it from the environment is preferable because it's not a part of the test configuration - I would want to use the upstream proxy settings of wherever I am running my tests.  (E.g., running Karma from my office requires different proxy settings than running it from my home, and those settings are specific to where the tests are being run, not what is being tested.)

**Comment by errorx666** [comment]
I called it a feature because it's doing something new that it wasn't doing it before, and a new flag was added to opt-out.
From another perspective, though, it is a bug fix.  It fixes the issue that Karma does not honor the system proxy of the environment it is running it.

File: lib/middleware/proxy.js [hunk 1]
```diff
@@ -50,6 +50,15 @@ var parseProxyConfig = function (proxies, config) {
       port = config.port
     }
     var changeOrigin = 'changeOrigin' in proxyConfiguration ? proxyConfiguration.changeOrigin : false
+    var useEnvironmentProxy = 'useEnvironmentProxy' in proxyConfiguration ? proxyConfiguration.useEnvironmentProxy : true
+    var agent
+    if (useEnvironmentProxy) {
+      var upstreamProxy = process.env[https ? 'https_proxy' : 'http_proxy']
```
Comment (dignifiedquire) [inline lib/middleware/proxy.js]: we should make sure to also read `HTTPS_PROXY` and `HTTP_PROXY` (not sure if this happens by default here)

**Comment by dignifiedquire** [comment]
@errorx666 sounds good, left one comment other than that looks good

**Comment by johnjbarton** [comment]
As this is more than two years old, please try a new PR against current master.

**Pull request closed without merging**